androidApp/src/main/java/com/mbta/tid/mbta_app/android/search/SearchBarOverlay.kt

```diff
@@ -0,0 +1,146 @@
+package com.mbta.tid.mbta_app.android.search
+
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.absoluteOffset
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.SearchBar
+import androidx.compose.material3.SearchBarDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.zIndex
+import androidx.navigation.NavBackStackEntry
+import com.mbta.tid.mbta_app.android.R
+import com.mbta.tid.mbta_app.android.search.results.StopResultsView
+import com.mbta.tid.mbta_app.android.state.getSearchResults
+
+@ExperimentalMaterial3Api
+@Composable
+fun SearchBarOverlay(
+    onStopNavigation: (stopId: String) -> Unit,
+    currentNavEntry: NavBackStackEntry?,
+    content: @Composable () -> Unit
+) {
+    var visible =
+        remember(currentNavEntry) {
+            currentNavEntry?.arguments?.getString("stopId")?.isBlank() ?: true
+        }
+    var expanded by rememberSaveable { mutableStateOf(false) }
+    var searchInputState by rememberSaveable { mutableStateOf("") }
+    val searchResults = getSearchResults(searchInputState)
+
+    fun handleSearch(stopId: String) {
```

Comment:
suggestion(non-blocking): `handleSearch` seems like a misnomer here; this seems to be something along the lines of `handleTapResult`?